app/adapters/parameter.js

```diff
@@ -0,0 +1,58 @@
+import DS from 'ember-data';
+import ENV from '../config/environment';
+
+const { JSONAPIAdapter } = DS;
+
+export default JSONAPIAdapter.extend({
+
+    caxe: Ember.inject.service(),
+    ajax(url, method, hash) {
+        hash = hash || {};
+        hash.headers = hash.headers || {};
+        return this._super(url, method, hash);
+    },
+
+    // Polyfill queryRecord
+    queryRecord(store, type, query) {
```

Comment:
queryRecord returns a promise. If a certain endpoint fails or 404 502 or whatever, we can catch the error and look in other places, like local storage or as simple as a pojo that's hardcoded somewhere in the app.

Reply:
this is an awesome idea